sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * The log configurations for HTTP messages.
+ */
+public class HttpLogOptions {
+    private HttpLogDetailLevel logLevel;
+    private Set<String> allowedHeaderNames;
+    private Set<String> allowedQueryParamNames;
+
+    public HttpLogOptions() {
+        allowedHeaderNames = new HashSet<>();
+        allowedQueryParamNames = new HashSet<>();
+    }
+
+    /**
+     * Gets the level of detail to log on HTTP messages.
+     *
+     * @return The {@link HttpLogDetailLevel}.
+     */
+    public HttpLogDetailLevel getLogLevel() {
+        return logLevel;
+    }
+
+    /**
+     * Sets the level of detail to log on Http messages.
+     *
+     * @param logLevel The {@link HttpLogDetailLevel}.
+     * @return The updated HttpLogOptions object.
+     * @throws NullPointerException If {@code logLevel} is {@code null}.
+     */
+    public HttpLogOptions setLogLevel(final HttpLogDetailLevel logLevel) {
+        this.logLevel = Objects.requireNonNull(logLevel);
```

Comment:
Maybe rather than require non-null, if null is passed in just set it to NONE. Be sure to document that though.